### ISSUE TITLE
Driver ignores maxRows value

### ISSUE THREAD

**Issue by xidex** [aws/amazon-redshift-jdbc-driver#19]
## Driver version
2.0.0.4

## Redshift version
PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.25813

## Client Operating System
Mac OS 10.15.7

## JAVA/JVM version
1.8.0

## Table schema
test_table(id VARCHAR(10), description VARCHAR(20))

## Problem description
When I set maxRows on the statement, i.e. `Statement#setMaxRows(int max)` method, driver ignores it and returns more rows than specified. v1 drivers work correctly, e.g.:
```
<dependency>
    <groupId>com.amazon.redshift</groupId>
    <artifactId>redshift-jdbc42</artifactId>
    <version>1.2.43.1067</version>
</dependency>
```

## JDBC trace logs
[redshift_jdbc.log](https://github.com/aws/amazon-redshift-jdbc-driver/files/6419965/redshift_jdbc.log)
[redshift_jdbc_connection_1.log](https://github.com/aws/amazon-redshift-jdbc-driver/files/6419966/redshift_jdbc_connection_1.log)

## Reproduction code
```
public void executeQueryWithMaxRows() throws SQLException {
    Connection conn = null;
    Statement stmt = null;
    try {
        Properties props = new Properties();
        props.setProperty("user", dbUsername);
        props.setProperty("password", dbUserPassword);
        conn = DriverManager.getConnection(dbURL, props);

        stmt = conn.createStatement();

        stmt.execute("CREATE TABLE test_table(id VARCHAR(10), description VARCHAR(20))");
        stmt.execute("INSERT INTO test_table VALUES ('1', 'First item')");
        stmt.execute("INSERT INTO test_table VALUES ('2', 'Second item')");
        stmt.execute("INSERT INTO test_table VALUES ('3', 'Third item')");

        stmt.setMaxRows(1);
        ResultSet rs = stmt.executeQuery("SELECT * FROM test_table");
        int retrievedRows = 0;
        while (rs.next()) {
            retrievedRows++;
        }
        rs.close();
        System.out.println("Number of rows retrieved: " + retrievedRows);
        //Assert.assertEquals(1, retrievedRows);
    } finally {
        if (stmt != null) {
            stmt.execute("DROP TABLE IF EXISTS test_table");
        }
        try {
            if (stmt != null)
                stmt.close();
        } catch (Exception ex) {
        }
        try {
            if (conn != null)
                conn.close();
        } catch (Exception ex) {
        }
    }
}
```

**Comment by iggarish** [comment]
Thanks for reporting issue. We will look into it and get back to you.

**Comment by iggarish** [comment]
We can reproduce the issue. We fix in the driver to mitigate this issue. The driver side fix will be available in 2.0.0.5 i.e. next release. We are planning to release within couple of weeks. 

Note that server side fix is better. We will discuss internally as it requires more time for it So right now driver will get more rows then max rows from server, but it will ignore it. As you understand, this is not efficient solution.

At the other end, if possible use TOP or LIMIT clause in SELECT statement.

**Comment by iggarish** [comment]
Released 2.0.0.5 with the fix.
